.github/workflows/upgrade-tests.yml

```diff
@@ -0,0 +1,72 @@
+# This is a basic workflow to help you get started with Actions
+
+name: "Upgrade Tests"
+
+env:
+  PERFIT_SERVER: https://perfit.dev.fedimint.org
+
+# Controls when the workflow will run
+on:
+  schedule:
+    # run daily during low usage hours
+    - cron:  '30 4 * * *'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  tests:
+    if: github.repository == 'fedimint/fedimint'
+    name: "Upgrade tests"
+    runs-on: [self-hosted, linux]
+    timeout-minutes: 60
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Prepare
+        uses: ./.github/actions/prepare
+      - uses: cachix/install-nix-action@V27
+        with:
+          nix_path: nixpkgs=channel:nixos-23.11
+          extra_nix_config: |
+            connect-timeout = 15
+            stalled-download-timeout = 15
+      - uses: cachix/cachix-action@v15
+        with:
+          name: fedimint
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+        continue-on-error: true
+
+      - name: Upgrade tests
+        run: |
```

Comment:
@dpc I skipped adding perfit since it looks like each workflow has a unique `--metric` identifier. Let me know if I should add before merging.

https://github.com/fedimint/fedimint/blob/2e0208f49dcf1105fc4bc0c97c16ef65cf3ae9c5/.github/workflows/ci-backwards-compatibility.yml#L64-L68

Reply:
Can be added after we migrate GH runners.

Reply:
Sounds good, created an issue to track: https://github.com/fedimint/fedimint/issues/5736